### ISSUE TITLE
Compatibility with Visualization Tool SeqtoSeq Vis /LSTMVis

### ISSUE THREAD

**Issue by mohammedayub44** [OpenNMT/OpenNMT-tf#478]
Hi,

I came across this Vis tool effectively shows some good visualizations for Sequence models. Looks like it uses model files form the other repo OpenNMT-py. Is there any way I can use this with the model files that are generated from this repo or out of box conversion to pyTorch models ?
Otherwise I'm guessing I have to run the same experiments again with OpenNMT-py. 

Thanks !

**Comment by guillaumekln** [comment]
Hi,

What do you want to visualize more specifically? OpenNMT-tf is not directly compatible with these tools.

**Comment by mohammedayub44** [comment]
Presently, I visualize the attentions from the exported model which works fine. For comprehensive debugging of sequence model - the topk words at each word interval and vector space of each word in the sentence allows to detect if the model has not trained enough or is it a beam search problem error etc like they describe on the video. This makes it very clear to our language experts as well while demoing it. 
I guess you could develop it from scratch but was leaning more towards reusing this repo and saving time. 
I hope that helps.

**Comment by guillaumekln** [comment]
These tools have a very good demo effect but it's not clear how they can be used for debugging: if a model is not performing well, the tool will not help in finding and fixing the root cause which usually comes from the data or the training procedure.

**Comment by mohammedayub44** [comment]
Sorry I meant troubleshooting the already learnt model, not the training process itself. However you are right, having some visual debugging tools while training might also be very helpful for fixing errors. 
So is there any motivation to include tools like this in OpenNMT-tf for already built models.

**Comment by guillaumekln** [comment]
> Sorry I meant troubleshooting the already learnt model, not the training process itself. However you are right, having some visual debugging tools while training might also be very helpful for fixing errors.

Yes, I actually referred to trained models. My point is if a translation is bad,  the visualization will not lead you to what should be fixed and the conclusion will usually be to continue training on more data.

> So is there any motivation to include tools like this in OpenNMT-tf for already built models.

It could be a good feature but I will not be able to work on that.

**Comment by mohammedayub44** [comment]
> It could be a good feature but I will not be able to work on that.

Fair enough 👍  I know on the other repo (Seq2SeqVis) lot of folks have already raised a PR to include OpenNMT-tf support. I'll just wait for it.
